Marlin/src/inc/SanityCheck.h

```diff
@@ -370,7 +370,21 @@ static_assert(X_MAX_LENGTH >= X_BED_SIZE && Y_MAX_LENGTH >= Y_BED_SIZE,
  * Filament Runout needs a pin and either SD Support or Auto print start detection
  */
 #if ENABLED(FILAMENT_RUNOUT_SENSOR)
-  #if !HAS_FIL_RUNOUT
+  #if ENABLED(FILAMENT_RUNOUT_MULTI_PIN)
+    #if FIL_RUNOUT_SENSORS >5 || FIL_RUNOUT_SENSORS <1
+      #error "FILAMENT_RUNOUT_SENSOR Number of sensors error"
+    #elif  FIL_RUNOUT_SENSORS >0 && !HAS_FIL_RUNOUT0
+      #error "FILAMENT_RUNOUT_SENSOR requires FIL_RUNOUT_PIN"
+    #elif  FIL_RUNOUT_SENSORS >1 && (!HAS_FIL_RUNOUT0 || !HAS_FIL_RUNOUT1)
+      #error "FILAMENT_RUNOUT_SENSOR requires FILAMENT_RUNOUT_SENSOR 0 to 1"
+    #elif  FIL_RUNOUT_SENSORS >2 && (!HAS_FIL_RUNOUT0 || !HAS_FIL_RUNOUT1 || !HAS_FIL_RUNOUT2)
+      #error "FILAMENT_RUNOUT_SENSOR requires FILAMENT_RUNOUT_SENSOR 0 to 2"
+    #elif  FIL_RUNOUT_SENSORS >3 && (!HAS_FIL_RUNOUT0 || !HAS_FIL_RUNOUT1 || !HAS_FIL_RUNOUT2 || !HAS_FIL_RUNOUT3)
+      #error "FILAMENT_RUNOUT_SENSOR requires FILAMENT_RUNOUT_SENSOR 0 to 3"
+    #elif  FIL_RUNOUT_SENSORS >4 && (!HAS_FIL_RUNOUT0 || !HAS_FIL_RUNOUT1 || !HAS_FIL_RUNOUT2 || !HAS_FIL_RUNOUT3 || !HAS_FIL_RUNOUT4)
+      #error "FILAMENT_RUNOUT_SENSOR requires FILAMENT_RUNOUT_SENSOR 0 to 4"
+    #endif
+  #elif !HAS_FIL_RUNOUT
```

Comment:
This code , seems to be loud , but is really well written and give goods warning results